service/matching/matching_engine.go

```diff
@@ -917,6 +923,73 @@ func (e *matchingEngineImpl) getDefaultBuildId(tqMgr taskQueuePartitionManager)
 	return "", nil
 }
 
+func (e *matchingEngineImpl) getBuildIdTaskReachability(
+	ctx context.Context,
+	nsID,
+	nsName,
+	taskQueue,
+	buildId string,
+) (enumspb.BuildIdTaskReachability, error) {
+	getResp, err := e.GetWorkerVersioningRules(ctx, &matchingservice.GetWorkerVersioningRulesRequest{
+		NamespaceId:                nsID,
+		TaskQueue:                  taskQueue,
+		DeletedRuleInclusionPeriod: durationpb.New(versioningReachabilityDeletedRuleInclusionPeriod),
+	})
+	if err != nil {
+		return enumspb.BUILD_ID_TASK_REACHABILITY_UNSPECIFIED, err
+	}
+	assignmentRules := getResp.GetResponse().GetAssignmentRules()
+	redirectRules := getResp.GetResponse().GetCompatibleRedirectRules()
+
+	// 1. Easy UNREACHABLE case
+	if isTimestampedRedirectRuleSource(buildId, redirectRules) {
```

Comment:
if a build is the source of now-deleted redirect rule, we should not declare it as unreachable, right?

Reply:
addressed